src/sortStudents.ts

```diff
@@ -1,16 +1,84 @@
+function averageGrade(array: number[]): number {
+  const result = array.reduce((sum, a) => sum + a, 0);
+
+  return result / array.length;
+}
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'NAME',
+  Surname = 'SURNAME',
+  Age = 'AGE',
+  Married = 'MARRIED',
+  AverageGrade = 'GRADE',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const newArr = [...students];
+
+  if (sortBy === SortType.Name) {
+    if (order === 'asc') {
+      newArr.sort((a, b) => a.name.localeCompare(b.name));
+    }
+
+    if (order === 'desc') {
+      newArr.sort((a, b) => b.name.localeCompare(a.name));
+    }
```

Comment:
to avoid [if hell](https://www.linkedin.com/pulse/avoid-if-else-hell-javascript-afsal-m-h/)

```suggestion
  newArr.sort((a, b) => {
  return order === 'asc' 
    ? a.name.localeCompare(b.name)
    : b.name.localeCompare(a.name)
})
```